google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -574,6 +578,31 @@ Timestamp getCommitTimestampOrNull() {
         : this.currentUnitOfWork.getCommitTimestampOrNull();
   }
 
+  @Override
+  public CommitResponse getCommitResponse() {
+    ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
+    ConnectionPreconditions.checkState(
+        this.currentUnitOfWork != null, "There is no transaction on this connection");
+    return this.currentUnitOfWork.getCommitResponse();
+  }
+
+  CommitResponse getCommitResponseOrNull() {
+    ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
+    return this.currentUnitOfWork == null ? null : this.currentUnitOfWork.getCommitResponseOrNull();
+  }
+
+  @Override
+  public void setReturnCommitStats(boolean returnCommitStats) {
+    ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
```

Comment:
Is it useful to check the connection state here? What breaks if this field is set or read on a closed connection?

Reply:
This also follows the standard of the other methods in this class, as well as the default behavior in JDBC, where all methods throw an exception if it is called on a closed connection. See for example [getAutoCommit()](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#getAutoCommit--).